pkg/controller/hawtio/hawtio_controller.go

```diff
@@ -402,30 +414,44 @@ func (r *ReconcileHawtio) Reconcile(request reconcile.Request) (reconcile.Result
 		}
 
 		// Mount client certificate secret
-		volume := corev1.Volume{
+		/*volume := corev1.Volume{
 			Name: clientCertificateSecretVolumeName,
 			VolumeSource: corev1.VolumeSource{
 				Secret: &corev1.SecretVolumeSource{
 					SecretName: clientCertificateSecret.Name,
 				},
 			},
 		}
+*/
+	//	deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, volume)
 
-		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, volume)
-
-		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+	/*	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
 			Name:      clientCertificateSecretVolumeName,
 			MountPath: clientCertificateSecretVolumeMountPath,
-		})
+		})*/
 	}
 
-	deployment.Spec.Template.Spec.Containers[0] = container
+	//deployment.Spec.Template.Spec.Containers[0] = container
 
 	// Create runtime objects
-	err = r.createObjects(objs, request.Namespace, instance)
-	if err != nil {
+	//err = r.createObjects(objs, request.Namespace, instance)
+	/*if err != nil {
 		reqLogger.Error(err, "Error creating runtime objects")
 		return reconcile.Result{}, err
+	}*/
+
+	// TODO(func Paramertes): refactor to avoid pass more params from function
+
+	hasUpdates, err :=r.reconcileResources(instance,request, r.client,r.scheme,isOpenShift4,openShiftSemVer.String(), openShiftConsoleUrl,serviceSigningCertificateVolumeMountPath)
+
+
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if hasUpdates  {
+		//return r.UpdateObj(instance)
+	} else if !hasUpdates {
+		//log.Debug("Reconcile loop did not find any resource changes to apply")
```

Comment:
Based on what I'm seeing in your PR, should this `Reconcile` function end here instead? Is there a reason it continues with object gets and logic after this point?

Should this section instead look something like this?
```golang
	if hasUpdates  {
		return r.UpdateObj(instance)
	}
        return reconcile.Result{}, nil
}
```

Reply:
fixed, still in process more cleanup and refactor the code after that.